tensorbay/client/tests/test_JobMixin.py

```diff
@@ -0,0 +1,193 @@
+        assert response_data == self.dataset_client.squash_and_merge._get_job(jobId)
+        open_api_do.assert_called_once_with("GET", f"jobs/{jobId}", self.dataset_client.dataset_id)
+
+    def test__list_jobs(self, mocker, mock_list_jobs):
+        params = {
+            "jobType": "",
+            "status": None,
+            "offset": 0,
+            "limit": 128,
+        }
+        open_api_do, response_data = mock_list_jobs(mocker)
+        assert response_data == self.dataset_client.squash_and_merge._list_jobs(
+            params["jobType"], params["status"], params["offset"], params["limit"]
+        )
+        open_api_do.assert_called_once_with(
+            "GET", "jobs", self.dataset_client.dataset_id, params=params
+        )
+
+
+class TestSquashAndMerge(TestJobMixin):
+    def test__generate_jobs(self, mocker, mock_list_jobs):
+        status, offset, limit = None, 0, 128
+        params = {
+            "jobType": "squashAndMerge",
+            "status": status,
+            "offset": offset,
+            "limit": limit,
+        }
+
+        open_api_do, response_data = mock_list_jobs(mocker)
+        job_generator = ReturnGenerator(self.dataset_client.squash_and_merge._generate_jobs())
+        assert [job for job in job_generator] == [
+            SquashAndMergeJob.from_response_body(
+                item,
+                dataset_id=self.dataset_client.squash_and_merge._dataset_id,
+                client=self.dataset_client.squash_and_merge._client,
+                job_updater=self.dataset_client.squash_and_merge._get_job,
+                draft_getter=self.dataset_client.squash_and_merge._draft_getter,
+            )
+            for item in response_data["jobs"]
+        ]
+        open_api_do.assert_called_once_with(
+            "GET", "jobs", self.dataset_client.dataset_id, params=params
+        )
+        assert job_generator.value == response_data["totalCount"]
+
+    def test_create_job(self, mocker, mock_create_job):
+        self.dataset_client._status.checkout(commit_id="commit-1")
+        title, description = "", "12"
+        source_branch_name, target_branch_name = "", None
+        draft_title, draft_description = "", ""
+        strategy = "abort"
+        with pytest.raises(StatusError):
+            branch_name = self.dataset_client.squash_and_merge._status.branch_name
+            self.dataset_client.squash_and_merge._status.branch_name = None
+            self.dataset_client.squash_and_merge.create_job(
+                title=title,
+                description=description,
+                draft_title=draft_title,
+                source_branch_name=source_branch_name,
+                target_branch_name=target_branch_name,
+                draft_description=draft_description,
+                strategy=strategy,
+            )
+        self.dataset_client.squash_and_merge._status.branch_name = branch_name
+        target_branch_name = self.dataset_client.squash_and_merge._status.branch_name
```

Comment:
here, we do not need to concern about the `_status`, just set target_branch_name equal to a specific branch name.